### DISCUSSION TITLE
Package names

### DISCUSSION THREAD

**Discussion by rcatlord** [tech-acs/crvsreportpackage#26] [Ideas]
Do you think we should change the package names? `crvsreportpackage` is sound but `crvs_report` is problematic because [CRAN doesn't accept underscores](https://cran.r-project.org/doc/manuals/r-devel/R-exts.html#Creating-R-packages) in package names.

Hadley Wickham has some useful guidance on [naming packages](https://r-pkgs.org/workflow101.html#name-your-package). Essentially, the name of your package needs to be unique, pronounceable and easily googleable. He and his Posit colleagues have form for choosing great package names like [lubridate](https://lubridate.tidyverse.org/) (for handling dates) and [forcats](https://forcats.tidyverse.org/index.html) (for relevelling factors). Can we improve on ours?

As a starter for ten, how about `crvstables` and `crvspipeline`? Please pop in your suggestions below:

**Comment by diego-ons** [comment]
I agree, I like those names, but I would add underscores (I like that naming style to be honest) so that it is:

- The package: `crvs_tables`
- The pipeline: `crvs_pipeline`

Happy to go with whatever, or not change.
What do you think @tbelay ?

**Reply by rcatlord** [reply]
Unfortunately, CRAN doesn't allow underscores in package names.

For inspiration, here's a helpful [blog post](https://www.njtierney.com/post/2018/06/20/naming-things) on naming packages.

**Reply by tbelay** [reply]
I agree that we should change the package name to something more more concise and descriptive name. Our work primarily focuses on the vital statistics (VS) component of the Civil Registration and Vital Statistics (CRVS) system. Therefore, I suggest naming it "vsreport". Even the title of the guide book mentions only 'vital statistics'.  
We can use the 'available' package to check if our selected package name is available for use on CRAN and other repositories.

**Reply by diego-ons** [reply]
@tbelay  are you happy checking what is available? We can do the transition before you go to Uganda

**Comment by tbelay** [comment]
All the suggested names are available except the validity of those with underscore

**Comment by jeroenminderman** [comment]
I second `vstables` and `vsreport` if they are available

**Reply by rcatlord** [reply]
Neither are on [CRAN](https://cran.r-project.org/web/packages/available_packages_by_name.html#available-packages-V).

**Comment by tbelay** [comment]
The ECA team had a discussion on it and we agree to collectively propose vsreport. 
@diego-ons @rcatlord @jeroenminderman @beardl-ons

**Comment by tbelay** [comment]
@diego-ons @rcatlord @jeroenminderman @beardl-ons 
If there are no other suggestions, I am going by '`vsreport`' and create the new repo in '`tech-acs`'. kindly let me know if that is ok with you.

**Comment by tbelay** [comment]
@diego-ons @rcatlord @jeroenminderman @beardl-ons @gulilattesfaye 
here is the link to the new repo:
https://github.com/tech-acs/vsreport
Let me know if you have problem accessing it.

**Reply by diego-ons** [reply]
I'm also about to create the `vstables` repo to move over the package.

**Comment by tbelay** [comment]
As discussed today, I have renamed `vstables` repo to `vspipeline`.
@rcatlord @diego-ons @jeroenminderman @gulilattesfaye @beardl-ons

**Reply by rcatlord** [reply]
Hi Tesfaye. Just checking that you plan to have two repos:

vsreport <- crvsreportpackage
vspipeline <- crvs_report
